### PR TITLE
[CI] Full build and test on Windows - weekly

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Run on Windows but on a weekly basis:
- Build on windows
- Test (verify maven goal) on windows
- If any failures then it will notify to the slack channel.

It can be executed on a PR basis with the GitHub comment: `windows tests` or using the UI with the new parameter `windows-ci` or GitHub label `ci:windows`

## Why

Simplify the default CI pipeline.

## Issues

See https://github.com/elastic/apm-agent-java/pull/1703#issuecomment-801141001

## UX

image

### CODE REVIEW

File: .ci/jobs/apm-agent-java-windows.yml
Comment (v1v) [inline .ci/jobs/apm-agent-java-windows.yml:31]:
```suggestion
            url: git@github.com:elastic/apm-agent-java.git
            refspec: +refs/heads/*:refs/remotes/origin/* +refs/pull/*/head:refs/remotes/origin/pr/*
            wipe-workspace: 'True'
            name: origin
            shallow-clone: true
            credentials-id: f6c7695a-671e-4f4f-a331-acdce44ff9ba
            reference-repo: /var/lib/jenkins/.git-references/apm-agent-java.git
```

**Review by kuisathaverat — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Comment by v1v** [comment]
@eyalkoren, if we merge this then you could use it to test https://github.com/elastic/apm-agent-java/pull/1671 in the CI. Though it will require to kick off a build with the specific refspec (see the screenshot in the description)

**Comment by v1v** [comment]
@SylvainJuge , this should answer your need to run for windows on a weekly basis that you asked for in:

- https://github.com/elastic/apm-agent-java/pull/1915#issuecomment-890789317

**Comment by SylvainJuge** [comment]
Thanks @v1v , that was exactly the PR I was thinking about :-).

**Comment by v1v** [comment]
@elastic/apm-agent-java , can we merge this?

**Comment by cachedout** [comment]
@eyalkoren I think this is related to https://github.com/elastic/observability-robots/issues/443 and we may be waiting for a review here. Would you mind taking a look and letting us know if we have a 👍 to merge? Thanks!

**Comment by felixbarny** [comment]
I have changed the goal to use `mvnw test` instead of `mvnw verify` as the `test` goal doesn't require a Docker setup.
That should unblock this PR.

File: .ci/schedule-weekly.groovy
Comment (v1v) [inline .ci/schedule-weekly.groovy:42]:
```suggestion
            booleanParam(name: 'windows_ci', value: true)
```

**Review by SylvainJuge — APPROVED** [review]

**Pull request merged**
